Follower terminology in the docs (#5067)
digital-asset/daml: docs/source/getting-started/app-architecture.rst

There are two important aspects here:

1. The data definition (a *schema* in database terms), describing the data stored with each user contract.
In this case it is an identifier for the user and the list of users they are following.
Both fields use the built-in ``Party`` type which lets us use them in the following clauses.

2. The signatories and observers of the contract.
The signatories are the parties whose authorization is required to create or archive instances of the contract template, in this case the user herself.
The observers are the parties who are able to view the contract on the ledger.
In this case all users that a particular user is following are able to see the user contract.

Let's say what the ``signatory`` and ``observer`` clauses mean in our app more concretely.

> A user Alice can see another user Bob in the network only when Alice is in the list of users that Bob is following (only if Alice is the ``following`` list in his user contract).

Reviewer: ```suggestion
A user Alice can see another user Bob in the network only when Bob is following Alice (only if Alice is the ``following`` list in his user contract).
```

> For this to be true, Bob must have previously strated to follow Alice, as he is the sole signatory on his user contract.

Reviewer: ```suggestion
For this to be true, Bob must have previously started to follow Alice, as he is the sole signatory on his user contract.
```

If not, Bob will be invisible to Alice.

Here we see two concepts that are central to DAML: *authorization* and *privacy*.
Authorization is about who can *do* what, and privacy is about who can *see* what.
In DAML we must answer these questions upfront, as they fundamentally change the design of an application.

The last part of the DAML model is the operation to follow users, called a *choice* in DAML.

.. literalinclude:: code/daml/User.daml
  :language: daml
  :start-after: -- ADDFRIEND_BEGIN
  :end-before: -- ADDFRIEND_END

DAML contracts are *immutable* (can not be changed in place), so the only way to "update" one is to archive it and create a new instance.
That is what the ``Follow`` choice does: after checking some preconditions, it archives the current user contract and creates a new one with the new user to follow added to the list. Here is a quick explanation of the code: 

    - The choice starts with the ``nonconsuming choice`` keyword followed by the choice name ``Follow``.
    - The return type of a choice is defined next. In this case it is ``ContractId User``.

>     - After that we declare choice paramteres with ``with`` keyword. Here this is the user we want to start following.

Reviewer: I always misspell "paramteres" the same way. 😃
```suggestion
    - After that we declare choice parameters using the ``with`` keyword. Here this is the user we want to start following.
```
